### DISCUSSION TITLE
STT: microphone not switched off after end of recording

### DISCUSSION THREAD

**Discussion by dguembel-itomig** [open-webui/open-webui#3942] [General]
# Bug Report

## Description

**Bug Summary:**
When using STT the recording is started and ended with the click of a button in the prompt (microphone icon) to start, "check" icon to stop". The recorded audio gets correctly transferred to whisper, however the microphone of the browser remains switched on. 

**Steps to Reproduce:**
0. Take docker image of open-webui 0.3.8 with local whisper as STT engine. 
1. In any chat, click on the microphone icon ("Record voice") and record a few words (Example: Say the numbers from 1 to 12). => The browser switches on the mic and records. 
2. End the recording by a click on the "check" icon at the same place.
3. Then, the recording gets transcribed into the prompt and the microphone icon (from 2.) re-appears BUT the micro of the browser is still switched on and remains so forever

**Expected Behavior:**
In step 2., when recording the audio is ended by a click on the "check" icon in the prompt, the microphone of the browser should also be switched off again. 

**Actual Behavior:**
In step 2., the mic remains switched on (forever)

## Environment

- **Open WebUI Version:** 0.3.8]
- **Ollama (if applicable):** [0.2.5]

- **Operating System:** Ubuntu 22.4.04 LTS
- **Browser (if applicable):** Chomium 126.0.6478.126; same with Safari under iOS 17.5.1

## Reproduction Details

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X] I have included the browser console logs.
- [X] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
+layout.svelte:88 Backend config: Object
+layout.svelte:119 connected
+layout.svelte:125 user-count Object
+page.svelte:23 Objectemail: "david.guembel@gmx.de"id: "29a2d62e-0898-43ca-ad77-4d2583e9d27d"name: "David M. Gümbel"profile_image_url: "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAASABIAAD/4QBMRXhpZgAATU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAZKADAAQAAAABAAAAZAAAAAD/wAARCABkAGQDASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9sAQwACAgICAgIDAgIDBQMDAwUGBQUFBQYIBgYGBgYICggICAgICAoKCgoKCgoKDAwMDAwMDg4ODg4PDw8PDw8PDw8P/9sAQwECAgIEBAQHBAQHEAsJCxAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQ/90ABAAH/9oADAMBAAIRAxEAPwDwBrL4FXXhiyn8UxSXmrPanyWglYFSR0O1lr0X4OeP/D3gTwimi+JtNMumuJBFIEDugbO3djJJHr9a/NXwjpOuNqttGzMEYhSwPIr6QZdf8o6M0rTxowKlW59cHFePXr1oNqnZp9Gjlq4eUHzbM9D03QfAmneKl1SSVb+2nkklWPG14kYnHQ9cH2NWNX+HegX2oXepaZq0dvZXx3LG/VXPbr3rwfWrvxhoeq7bS2aWZ0yA+AMD3ri774ja6+inSZrYW92s5k8wHlcHp/8AqrypUMX7PlU0l6HC8FKU+ZT+XQ+vPD/w08U3usWmiahqy2mnQguuGCs6gcAc969D0PRfEmjxQX7+H2hfzTCsrKWlkU5wSBnAxz1r43+Gl54r8W3j3Oo6xLLLAvybpCoQe2K+8oviJreh+HbSCDVinAjyyb8t06055fiJqN2ncv8As6T3nbucb4ws9budJuLyyTyLqI7oYfLOGI6g8ZwfWoYLeWOzkh1ON1W7RApQlRH3YAN1r37wna+NPGElulmrXKMoMkzw7I1z7kc/Su98QeAtK0qyaTUALm453HoAcdsV9Hw9wZUrL96kjgxtTDYZJyd2fHLSRwyHTkwyydQ3UCruo6hBosdtcxbcxrjnoM0l/Z6drN7OIW8i4t5CAw4BAPSvL/F0Wo3VvPazShRH91s4z+NcNLBxc3Sas1uddGoptWR7tpXxYv4EjiW5W3kXooO0HPeqmpePbu+uF/tNVnd+AW7j2rzbw/8ABa68TWFrqb6qxaFAX2nofavbdA8Ga74btAl7psfiKwHIDDEoFeksqbi5U1ex3OvGElCpK1+50WifFG1s9Ojtha7dnGA3H8xWt/wtu2/59z/31/8AXrjz4j+Gtm7QXXhm8tZFPMYRiB9McUf8Jb8Lf+gDe/8Aft68l0KqduV/gekrd0f/0PgX4W6S+q3l5G8Y83G+MZ6KOuD617PceHW8PW0+qSTD99BuRDy2/wBya8w+FIlsvFVtGUZlmUqxH3RkZ5PTtXceMIzf6qdGN21pBv4Yng4PrXj14RUXMivecWrnL/EPxFNqun6YEgS3W3ASWXdgkk4GT6V5lqHgafUbqO6glSWBsA4bnca9Z8VeB9Tt0giUC6s59qk/3s4yfwrkINCtbZ7q304sptiD1J5HOBXLRrSrRTg9DijTlCR3Hw8+GbaPc3MmpJKoKgp5R4NfoJ8NNI0X+zPPezWWGFBxKoPz+vNfm94c8U62+s29pf3DrG5KAZPPp1Ffoz8IbDXV0CY3bBrRuVB+9n3r7LhXCKWKgqkdEedxLi5UsHOVKWp9WWniUQ6bHaWkaRDYMbABj8q4PW7eXUYn8/LIzZOKv6NbF44QBwV5r0BtG8uzj3r989K/V+WNNtRVj8+wUp10pTdz4t1/wDpem6s1+GEUchyyHjJNc14l+GGheJkia4LQRLxmPgH8a6D9oG8uv7dTw/bwy2YC7xcsCEYegPQ145/wtq+0/wAOjwqb5ZJkkByRyRnP3q/M84xNCniJ3gz9iyLh/EV6NOpRqxTfS9mvW57V4V8ADwpbrZ6ZNJNDKRndzgV9XeF/DsMdtCWTcxxkYrV+C3h3RbnwhpWuTuL8XkSszdcEgV9DT+C7K1tF1CxxsI6eldUcZTjBRhofM5tl1WdVzk72Z5BN4Q0O4YST20Jcj+JRmov+EK8Pf8+sH/fC1y/ir4v/AAw8J61No3ibxBa2V9FgmKSZVYKenBNc7/w0H8EP+hqsf+/6f41neHVnm+3qdEz/0finwPdXuhXplSLz7CU715y/I6DP+NJ8Uri6C2808DwAtnHcZ6GvK5fiPPoCWNlaOslvDwXKkEY/xxXTfFHxzZax4P0vxHBMTNMwWSI9gK8alGq3yVFp0ZjCLSsfROha5o954LhsJJG+0Io8t5O7emTXmvwv1HSbPx5c2niFXedJdywY+VwR1NReDNFmvPA0Ov3N0q2zr5kQzyuPWsW68RRrOl3pNm0t7AC0lyP7g688etcmT5ZPDzftNUzL95fU+m9W8O+GrbWbC6uIkRr24MkcajnJ6CvrT4dTWGhaXPY69qlvbNMS0azTJFx6AMR0715R8IPhte6vpVp4l8Z3hjMsSSRqAUMETbWGHUEmUtjAJG0lSOeBJ4x8B2hvvtNrbKrKNsXnZfC44ITJ3PwCcDIHevp8uz2WGq3jC9u7OzG8MLFU1GpOyvdpdV69D7h8L6KfLjLbdq9wa76W2imkVIG5XpX5ZWV34z8CTxajoN/cObUqkiMRGCi4yChHcccBTntX214N+MeieI/DlrrDbre5cAPGf7wHJU91znGefavsst4op4qbjJcsvXc+YxfC8sFTTg7x/I9T1PwRpHiLKeI7eK5hXOFdQf515d/wqn4K6hqtlo2o6FakwSfu2CADn1wP51y/iT4p373ckOnybQgODuxk15ro+v8Aip9Qe7u4QVL7/NJOBzxXj5zmdOFVRV3frYvC1+WOp+ofgzwT4c8N6YumaNaxw2Ea/IiD5V9gK534heN9L+HXhXUtevomeGyhkkKLyTtBxgV478P/ABZ4j0cE+I75JNPkXep3Y2+3OK+U/wBo79pG21Y3fh7wei3MF0hgnnZwQPUqvPbivKxWIUI+0k7HrU60eWx+V/iXx34n8UeJ9a8SavYiS41O9nuPmGSqO3yJ0/hXA/Csb+3tT/6Bq/l/9avUNV0MNchoD5iFRyCB+FZv9gzf3D/30K+Wc1L3m9zVKpb4T//S/K/x74B8WeHJjb6vY+em4YmtzvibuPmwMfQ4rl7LwT4q1y1FpGgSAHIV2Jx+FfrZd6Npurow8tCGGMEcHPqK8v1T4cwabvl0uFbZzlgVUlDn27fhU1aU1rDY6quFa2PmX4ffC/xbbQw6dqt+507tAAduPb2/CvVtb+EsNl4bujaXguLmR0igtwCSTM6oFwvJI3Z6Zr3/AOGdja6fp902oX0UmofdSKUAYHqoPX8K9j0e20LwzNF40161jk+yyxC3DIGXznb5X2nqUxuAHfnPHOdaooQcpO5w4VynWjStZnTz63B4D0XwV8N/F7PFrN7p3mShRvUParFFKDISy5TzVVWI+b5iSOtSeKN9vaXU1heiKK4UKwcMcHgZALABs44IAPAwcV5d8WfiDpvibw7oXxZtTtj8P3l3p1wGUIoimdDuyx53TCNRwTyeMjNZ2r+KNQ0bR7fXJ/MvraSECSGKPe6qcHc5LAKAhye49DmvPnJVIqcFby/4e59k5csnF7HO682qadFM8jGOO4Aj3qE+YgFyR1BwDnp6n1ra8BzmTw3qkFnEziK5iljyckBg6uc9/mIz6nnArz3VBca5pSXdmx+xXBZQjN+7dGIRiCf9YeBtAIGeOMHPc/DXR/EVr4f1yeCzkVYEWKJwpG9huwAuASHYKoIz83XGRXFHEOE1LYwx+G9pQnFK+h0GtXMtvFiYmC6EeDxlge1cRH8W/GGk2MaCKK5aFsNu+UlB6+9c/NrksrrM7MRccbmOfm/Gse5srTxJpN9Gs3lTRjlG+Vjg+3UVs8TN1PaSk7f1uj84wsqN/fM3VfHnxD8a3l5aahqUkVlcjEUMWAie2VAry7VNJ1yC/wDKgLOlooEgKkYz157V6t4W0a70JYIjp1wzMzEFmBjyDxjPI/zzXXeI30+1gFxcMnmXBPmDOPmHb3rhrZjKWKjDkcl36I9eM4e0UVHQ+fbOTw1bRGG7ad5gx3ENxn0HI4FW/tnhH0n/AO+v/r1bk0y++0Ttawq0bOWGOcZwcUn9mat/z7ivobRPcVj/0/LV1DxV4KkQ6xbNJZ8ETKC0eP8Aa7r+Nen+GvFGieJolhysEjcbXPfPY9DXUW8mgT2SXniiVmu55OIeDGE/2h6fWvL/ABT8NrG9vmvvBZNhK+WCL/x7NgdgPu/Uce1cNLHukrt6GtLN4S92to+502t+CYLndJaH5hnBA9DwRXm9x49XVPDt5pOpX7yQWM8AtmIZjk7hIVORuAjDuNx5xgYHJpnxz4z8GwXGi6/bsjvFIsMjfMpIU4ZH6HHXH5ivCNN0C98Ri40OzSSa4vECxvGDmKR2ADbsccErk9S3GTX3WSZXh8bh6tScbdPLuc2YYydGrT9m79T3fTBe2vhHxh4Z1j/iZ+HtUZ722uIWJBWRURZYnUg5iZEPALKCxwdmB5p4Fj8RzXU9h4h1YnS9OYxC1Rx50sOd0aMVLBGY4Z2BBUDIPCgfcOv/AAwHh79n3R/g7q1zEdd8k3t7KnmhVH2gXc8StE2/aNzRllJ3KDlSD5Z/MuPSLdLlrG2vGja2OJRA6FhKhIwNpZu3ysoyDnjGK8TIuHqOL9pGhW5lF9vO39M9DMc1r0XF16XK353/AOG9Oh92aNcaT/adlqWtiJrsEpaW6ACKBMDGFUsERRtBI5JOdzHJr2Xxf8X9C0v7Fp9nFNDDM+2WUxhI3Ean5QxyGHIJI74GeGx+dGlW72bWZFzcrHbec0kRmLiWR0IPDH5QG2tnrx6HFdnHe6nrDsmtyteWiMjMm7gncApxwcqeMD+90IFehW8OKtd357NbHPLi1woSo01bm37nth06DV7W/wBY014004+Z5IwPlwSQcDGPpXC3Fx4fitDfzov2mGLCsCRuJ9faqnh7V/7Pur3TYlilsmkAkgJaMDI5w2Wznv8AKB2BrsZtGg8Y6YkU9pBpdxY8l0wVZFGdp749yK+RzXhvGZe1PEL3L2TWqv8AmfC88uZvm3KGleJBcafb6Le3cSX8iec+0g5UdByfwqWz8M6T8WrB7K7lfR7vTmkaSRclM/wjHqa47TvAUs/hWbxVq629lulZRIr4DBXwpz2BGK9s0a2j8F6VdX2qK12ksYZzbsNocrxknjAGOTXjyko/A9T6jKsNzPmj0ep8rNot/pk0tnFrihUdsbzsbjjkH6Un2XUv+g7F/wB/K9mudY8HeP521rxVJFZ3kW22VGhwTHEBtb5eOcn8ag/sD4U/8/1v/wB+mr1FX73+49/2T7L7z//U4XVNX0sXYnmvo2jK7mYMDj2xXb+DtWiudPF1ZyP9mDEs0ikJt6DGexry+1+A2i2HiWLxBPcm70O+Ktb4bNu5Y87vT2HTNdt4qikuNQs/Bfht/Je5IRinASLOGP8AhXmYjIuWfs655lTCwa3uji/jBqOLGz0lZFZLtzOWU5BReFUezHPftX0F+yp4G0qaNfFVyxn+zu724SPEUTopjkkG0AtiQlTwBxnJ4A+TvihdWN94iOm6KhhstDjWwiP/AD0NuT5jdeQXZivquCa7v9nzxbrthq0mi6nq4GlSwGGNLqQoyBWWUlSmDkhCg5Od3ccH9BzjK6+DyeFOjtvJ9ddvzsb8PVKM8W+ffofWHja+/sPxTq/jC+lSOHQ9OADYMe+VxDJMEXGRhkbP+9n1J/MzWLyDUNVk1HwfqIm0a/8An8oRxrNau5y0JVH3+WgIIfaoAyo3bdx9e/ak+K+p+KdEuPCfhS4N5FKcXV6nClg25tvyguS/Jboo+XOBmvnHwfo11bWItLibb5gjLgkjBXIDbcHBGSAevc15XAGAxUKjqUtIvft+h7fFmLoTioT3Wx6LFdTMf+JlDGXXq4Xkcc5GOM5/Slk1KZYUWNt0TNHyvJVQwG45yRx1xjgdqh8t4AJF6xHORkg59Vx0B5H+FZc9y63ck8o8oAMMqQVAxliOvXjk8jvX7VOfKrs/OVrsdRDctJILmIZZ2+bPXj268nJrsx4ou7Lw5dLaWzSXl+v2W3YDGSfvHceF2A5yfX6Z9J+FP7PeueJdMOra7/xL4tpaK0k+WR8fxOc9+uB2PcGvTte8N6XBpcXhDxtocdxptuD5LRr5LxhuTskjxkE9VYMue3evzfjLNlVofV403Z2d3t8v6R7uAyb2nvOSuuh4vpel+C9Q8JwfD+yvJLnVdPRHw0jKqu3c9mAI47f09X8G6FayCXS9ent7WO3jMtzFdOW3yFdq4T+IHr+XFcwPhpEL6K+8J6it1Km7aJgI50RceXGFXIkyTgkHpzgDp2vxJ0nTH8I+HdRt9WB1xiXmjcrmNCCMDb1APGCa/HsRWdKfvO0d77f16HbQ9thpctVaM8W8R6d4JstR+yWOmai6xIodowm1n7ke3TFYPkeFv+gTqn5JXr2peItHspoxfy5lljRz5agLjG3p68Vn/wDCXeGv+ekn5CvDfEuHi7Kb+8JcQWdlE//V8/8A2VvEep6tdzeCNUcXekTRM/lSDdsP+ye2ah+Ickngb4gXOn6E7CO5TBMhLsgXkBG4I/Wsf9kX/keP+2Df0rV+On/JTh/utXt4vXL6Mnve3y7F1Ir6xUXSx4xYENp0bEDKojA89V4/r+lTX8eW4JUS5DBeAceoHHPp09qg07/kGJ/1yX+dW77rH9W/mK/dHBONmj8+hJp3Rz32G3eRkC7fMYBiMZwFQgZx0G7p04HHFaEZCQRxqoCZC45PDMVP6LTE/wBeP98f+gRU5f8AVx/76f8AobVnRpximoqxU5N7sz7qV4jwd28svzc4wM5B9c1594l1C5c6TETgTyFnxkZ2cD8M849ea72/6x/77f8AoNea+I/9bon+9J/Ovm+J5NYOrbt+p6mUK9eFz9kPgTr2oa34JsE1FhK0EaIr4+fAC45/GvSNRtba+Zre7jEqZx83PGGNePfs5f8AIm230X+S17RN/wAfLf739Grxcu/eYT95rp11O+t7td8uh81eM9CsNEnc6erJ82MZ4A+lc9Na2+tW1r/acYuJGdohKw/eKo2gYbqR7HK+1d18Sv8AXN/v1xtl/qLL/ru/81r8ozKhCTnCUU126H2sEpU0pani/iCaTTtVms4DlIyQNwyeprF/tS6/2fyrV8X/APIfuv8AeP8AM1zNfytjcFR9tP3Fu+i7nwdXSTSP/9k="role: "admin"token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjI5YTJkNjJlLTA4OTgtNDNjYS1hZDc3LTRkMjU4M2U5ZDI3ZCJ9.ZuAX_1QdY9T3Y9q2htW6Qo1l6n-IiNDm9EYebLRHgBM"token_type: "Bearer"[[Prototype]]: Object
websocket.js:43 WebSocket connection to 'wss://leg.emeto.de/ws/socket.io/?EIO=4&transport=websocket&sid=7GajH6qAACtotEIHAABO' failed: 
doOpen @ websocket.js:43
1 weiteren Frame anzeigen
Weniger anzeigen
index.ts:64 Array(13)0: {id: 'codestral:latest', name: 'codestral:latest', object: 'model', created: 1721220517, owned_by: 'ollama', …}1: {id: 'dg-infra-expert', name: 'DG Infra Expert', object: 'model', created: 1720810336, owned_by: 'ollama', …}2: {id: 'dg-sekretaer2', name: 'DG Sekretaer2', object: 'model', created: 1720809733, owned_by: 'ollama', …}3: {id: 'dg-sekretaer3', name: 'DG Sekretaer3', object: 'model', created: 1720897309, owned_by: 'ollama', …}4: {id: 'dsb-itsec', name: 'dsb-itsec', object: 'model', created: 1721216369, owned_by: 'ollama', …}5: {id: 'gemma2:27b', name: 'gemma2:27b', object: 'model', created: 1721220517, owned_by: 'ollama', …}6: {id: 'gemma2:latest', name: 'gemma2:latest', object: 'model', created: 1721220517, owned_by: 'ollama', …}7: {id: 'infra-expert', name: 'infra-expert', object: 'model', created: 1721046146, owned_by: 'ollama', …}8: {id: 'llama3:latest', name: 'llama3:latest', object: 'model', created: 1721220517, owned_by: 'ollama', …}9: {id: 'mistral:latest', name: 'mistral:latest', object: 'model', created: 1721220517, owned_by: 'ollama', …}10: {id: 'mixtral:latest', name: 'mixtral:latest', object: 'model', created: 1721220517, owned_by: 'ollama', …}11: {id: 'otrs-expert', name: 'otrs-expert', object: 'model', created: 1721119408, owned_by: 'ollama', …}12: {id: 'phi3:medium-128k', name: 'phi3:medium-128k', object: 'model', created: 1721220517, owned_by: 'ollama', …}length: 13[[Prototype]]: Array(0)
VoiceRecording.svelte:165 Recording started
VoiceRecording.svelte:171 Recording stopped
VoiceRecording.svelte:143 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12,

**Docker Container Logs:**

INFO  [apps.audio.main] file.content_type: audio/wav
75bafba2-d47b-4701-ba54-62f40442a0c2.wav
INFO:     172.18.0.1:46426 - "GET /ws/socket.io/?EIO=4&transport=websocket&sid=SMpJU6oPjd0VUBArAAAE HTTP/1.0" 200 OK
INFO  [apps.audio.main] Detected language 'de' with probability 0.995208
INFO:     127.0.0.1:35490 - "GET /health HTTP/1.1" 200 OK
Unknown session ID IHJGYopDkktTSdARAAAB disconnected
Unknown session ID _pMqYmielJUqdT7KAAAD disconnected
INFO:     172.18.0.1:46474 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P30hh-x&sid=SMpJU6oPjd0VUBArAAAE HTTP/1.0" 400 Bad Request
INFO:     172.18.0.1:46486 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P30hh-x&sid=r40V1ebiVXkAdg24AAAA HTTP/1.0" 400 Bad Request
Invalid session SMpJU6oPjd0VUBArAAAE (further occurrences of this error will be logged with level INFO)
ERROR [engineio.server] Invalid session SMpJU6oPjd0VUBArAAAE (further occurrences of this error will be logged with level INFO)
INFO:     172.18.0.1:53518 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P30hs-7&sid=SMpJU6oPjd0VUBArAAAE HTTP/1.0" 400 Bad Request
INFO:     172.18.0.1:53528 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P30hs_N&sid=r40V1ebiVXkAdg24AAAA HTTP/1.0" 400 Bad Request
INFO:     172.18.0.1:53540 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P30htC- HTTP/1.0" 200 OK
INFO:     172.18.0.1:53546 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P30htDz&sid=sNbL41zOONZdc-mKAAAG HTTP/1.0" 200 OK
INFO:     172.18.0.1:53562 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P30htDz.0&sid=sNbL41zOONZdc-mKAAAG HTTP/1.0" 200 OK
INFO:     172.18.0.1:53584 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P30htJJ HTTP/1.0" 200 OK
INFO:     172.18.0.1:53608 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P30htKs&sid=Pisi7G0s2cDz-YxPAAAI HTTP/1.0" 200 OK
INFO:     172.18.0.1:53596 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P30htKv&sid=Pisi7G0s2cDz-YxPAAAI HTTP/1.0" 200 OK
INFO:     127.0.0.1:59618 - "GET /health HTTP/1.1" 200 OK
{'text': '1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12.'}

**Screenshots (if applicable):**
![image](https://github.com/user-attachments/assets/8aa7c719-f0ce-4060-8158-ebeae7c2a3a6)

## Installation Method
Docker package as described, separate packages for open-webui and ollama. 

## Additional Information

None necessary I suppose, but please feel free to ask me for anything you feel might be useful. 

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by dguembel-itomig** [comment]
@tjbck if I may be of any assistance, please let me know. I do think this is a bug (microphone should be switched off after audio recording was stopped; it's a security problem and also energy consuming (smartphone..) to have it switched on unneccessarily all the time)), but I do not feel comfortable trying a fix myself, sorry.

**Comment by dguembel-itomig** [comment]
Created with the help of claude-engineer[1], please find attached a diff that might fix the problem (or at least give a hint about why it's not working as desired). Essentially, it stops all tracks so the microphone of the browser should get properly switched off when stopping a recording.

What it does is:

 1 Added a new mediaStream variable to store the MediaStream object.                                                                                                                                    
  2 Updated the startRecording function to store the media stream in the mediaStream variable.                                                                                                           
  3 Modified the stopRecording function to stop all tracks of the media stream, clear the mediaStream variable, and set recording to false.                                                              
                                                                                                                                                                                                         
These changes ensure that when stopRecording is called, all tracks of the media stream are stopped, effectively turning off the microphone.  The mediaStream is then set to null to clean up the reference, and the recording state is updated to reflect that the recording has stopped.                                                                              

Disclaimer: I do not know how to locally rebuild open-webui with that change and then run it - so I was not able to test locally. Still, I thought this might be helpful. 

[VoiceRecording.svelte.diff.txt](https://github.com/user-attachments/files/16272973/VoiceRecording.svelte.diff.txt)


[1] [](https://github.com/Doriandarko/claude-engineer)

**Comment by thiswillbeyourgithub** [comment]
Still happenning in 0.3.13

**Comment by tjbck** [comment]
Should be fixed on dev!

**Comment by dguembel-itomig** [comment]
Hi! Thanks for the update. AFAIK the fix is in this commit [1] and thus included in the 0.3.16 release (please correct me if I am wrong about this). 

I tried again with current Chromium under Ubuntu 24.04 LTS (chromium v 128.0.6613.84) but unfortunately the problem seems to persist. I'll investigate if I manage to find why… 

[1] https://github.com/open-webui/open-webui/commit/153a2876b430f423a34b2b0055fbbd34505b132f

**Reply by mardet** [reply]
I just tested this with the latest version v0.3.30 and I confirm the problem still persists.

**Reply by dguembel-itomig** [reply]
Yes, re-tried also with latest docker image - problem still there. Chromium on Ubuntu 24.04 as well as latest Safari. How can I help to debug it?

**Reply by Mushy-Snugglebites-badonkadonk** [reply]
Yes, was happy to see the microphone remaining open issue resolved for calls on Safari, but the issue still persists when using STT manually via the OWUI microphone button. 

Let me know how I can support! .i.e., bug reports

**Reply by maxkuminov** [reply]
The issue still persists for me in v0.3.32. I have tested Firefox in Linux, and both Firefox and Chrome on Android.
